mediator/med_map_mod.F90

```diff
@@ -594,529 +587,629 @@ logical function med_map_RH_is_created_RH1d(RHs,mapindex,rc)
-                if (chkerr(rc,__LINE__,u_FILE_u)) return
-                call ESMF_FieldBundleGet(FBSrc, fieldName='Sa_v', field=vsrc, rc=rc)
-                if (chkerr(rc,__LINE__,u_FILE_u)) return
-                call ESMF_FieldBundleGet(FBDst, fieldName='Sa_u', field=udst, rc=rc)
-                if (chkerr(rc,__LINE__,u_FILE_u)) return
-                call ESMF_FieldBundleGet(FBDst, fieldName='Sa_v', field=vdst, rc=rc)
-                if (chkerr(rc,__LINE__,u_FILE_u)) return
+          ! Loop over the fldsSrc types
+          do ns = 1,size(fldsSrc)
 
-                call ESMF_LogWrite(trim(subname)//" --> remapping "//trim(fldname)//" with "//trim(mapnames(mapindex)), &
-                     ESMF_LOGMSG_INFO)
+             if ( fldsSrc(ns)%mapindex(destcomp) == mapindex .and. &
+                  trim(fldsSrc(ns)%shortname) == trim(fieldnamelist(nf))) then
 
-                call med_map_uv_cart3d(usrc, vsrc, udst, vdst, RouteHandles, mapindex, rc=rc)
+                ! Determine mapping of indices into packed field bundle
+                ! Get source field
+                call ESMF_FieldGet(fieldlist_src(nf), ungriddedUBound=ungriddedUBound, rc=rc)
                 if (chkerr(rc,__LINE__,u_FILE_u)) return
+                if (ungriddedUBound(1) > 0) then
+                   do nu = 1,ungriddedUBound(1)
+                      npacked(mapindex) = npacked(mapindex) + 1
+                      if (nu == 1) then
+                         packed_data(mapindex)%fldindex(nf) = npacked(mapindex)
+                      end if
+                   end do
+                else
+                   npacked(mapindex) = npacked(mapindex) + 1
+                   packed_data(mapindex)%fldindex(nf) = npacked(mapindex)
+                end if
 
-                used_cart3d_for_uvmapping = .true.
-             end if
-             CYCLE
-          end if
-       end if
+                if (mastertask) then
+                   write(logunit,'(5(a,2x),2x,i4)') trim(subname)//&
+                        'Packed field: destcomp,mapping,mapnorm,fldname,index: ', &
+                        trim(compname(destcomp)), &
+                        trim(mapnames(mapindex)), &
+                        trim(packed_data(mapindex)%mapnorm), &
+                        trim(fieldnamelist(nf)), &
+                        packed_data(mapindex)%fldindex(nf)
+                end if
 
-       ! -------------------
-       ! Get the source and destination fields
-       ! -------------------
+             end if! end if source field is mapped to destination field with mapindex
+          end do ! end loop over FBSrc fields
+       end do ! end loop over fldsSrc elements
 
-       call ESMF_LogWrite(trim(subname)//" --> remapping "//trim(fldname)//" with "//trim(mapnames(mapindex)), &
-            ESMF_LOGMSG_INFO)
+       if (npacked(mapindex) > 0) then
+          ! Create the packed source field bundle for mapindex
+          allocate(ptrsrc_packed(npacked(mapindex), lsize_src))
```

Comment:
It would be nice to check those are allocated or not. It is valid for all allocate statements.

Reply:
This allocate just occurs at initialization - so Its not clear to me why it would not be allocated. But I can add a check statement.

Reply:
But will this require rerunning all of the tests again?

Reply:
it is fine, we could do it later. I am testing it with HAFS now.

Reply:
Okay - that would be great. Maybe you could raise an issue for this as well so that we don't forget to do it. Thanks for pointing this out.